### PR TITLE
Refactor postgresql scaler config

### PR DESCRIPTION
**Pull request by dovics**
### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #

Relates to https://github.com/kedacore/keda/issues/5797

### CODE REVIEW

File: pkg/scalers/postgresql_scaler.go [hunk 1]
```diff
@@ -42,12 +41,46 @@ type postgreSQLScaler struct {
...
+	Port     string `keda:"name=port, order=authParams;triggerMetadata, optional"`
+	UserName string `keda:"name=userName, order=authParams;triggerMetadata, optional"`
+	DBName   string `keda:"name=dbName, order=authParams;triggerMetadata, optional"`
+	SslMode  string `keda:"name=sslmode, order=authParams;triggerMetadata, optional"`
+
+	Password string `keda:"name=password, order=authParams;resolvedEnv, optional"`
+}
+
+func (p *postgreSQLMetadata) Validate() error {
+	if p.Connection == "" {
```
Comment (SpiritZhou) [inline pkg/scalers/postgresql_scaler.go:61]: Maybe it's better to return directly if p.Connection != ""

File: pkg/scalers/postgresql_scaler.go [hunk 2]
```diff
@@ -83,66 +116,26 @@ func NewPostgreSQLScaler(ctx context.Context, config *scalersconfig.ScalerConfig
...
 	}
 
-	meta.activationTargetQueryValue = 0
-	if val, ok := config.TriggerMetadata["activationTargetQueryValue"]; ok {
-		activationTargetQueryValue, err := strconv.ParseFloat(val, 64)
-		if err != nil {
-			return nil, authPodIdentity, fmt.Errorf("activationTargetQueryValue parsing error %w", err)
-		}
-		meta.activationTargetQueryValue = activationTargetQueryValue
+	if !config.AsMetricSource && meta.TargetQueryValue == 0 {
```
Comment (SpiritZhou) [inline pkg/scalers/postgresql_scaler.go:126]: Many scalers are checking AsMetricSource, would it be possible to simplify it and move this logic to Validate()? @wozniakjan

Comment (wozniakjan) [reply pkg/scalers/postgresql_scaler.go:126]: valid point, although I am currently lacking good ideas on how to do that elegantly so I'm ok with either. Some amount of duplicate code in the scalers to check `AsMetricSource` or having generic way somewhere in the `TypedConfig` are both ok to me.

File: pkg/scalers/postgresql_scaler.go (hunk 1 above)
Comment (wozniakjan) [reply pkg/scalers/postgresql_scaler.go:61]: I agree, that might be a little bit nicer, although two nested `if` statements are imho still within reasonable boundaries of readability so I'm ok with either :)

**Review by wozniakjan — APPROVED** [review]
lgtm, thank you!

**Review by zroubalik — APPROVED** [review]

File: pkg/scalers/postgresql_scaler.go (hunk 2 above)
Comment (SpiritZhou) [reply pkg/scalers/postgresql_scaler.go:126]: What about back to this solution(https://github.com/kedacore/keda/pull/5819), which can make it more readable if moving config checking in Validate()?

**Pull request merged**